sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowTablesSuite.scala

```diff
@@ -39,7 +39,7 @@ trait ShowTablesSuite extends SharedSparkSession {
   protected def runShowTablesSql(sqlText: String, expected: Seq[ShowRow]): Unit = {
     val df = spark.sql(sqlText)
     assert(df.schema === showSchema)
-    assert(df.collect() === getRows(expected))
+    assert(df.collect().toSet === getRows(expected).toSet)
```

Comment:
After modifications in `show tables with a pattern`, the order of rows was changed.

Reply:
how about `checkAnswer(df, getRows(expected))`

Reply:
Replaced it by `checkAnswer()`